AsyncDisplayKit/ASCellNode.mm

```diff
@@ -204,6 +244,26 @@ - (void)__setHighlightedFromUIKit:(BOOL)highlighted;
   }
 }
 
+- (NSIndexPath *)indexPath
+{
+  ASDisplayNodeAssertMainThread();
+
+  UIScrollView *scrollView = self.scrollView;
+
+  ASDisplayNode *owningNode = scrollView.asyncdisplaykit_node;
+  if (_scrollViewType.isTableView) {
+    return [(ASTableView *)owningNode indexPathForNode:self];
+  } else if (_scrollViewType.isTableNode) {
+    return [(ASTableNode *)owningNode indexPathForNode:self];
+  } else if (_scrollViewType.isCollectionView) {
+    return [(ASCollectionView *)owningNode indexPathForNode:self];
```

Comment:
`scrollView` would return ASCollectionView or ASTableView, we need the owningNode which is the corresponding ASCollectionNode or ASTableNode.

Aside from that I removed support for ASCollectionView and ASTableView as requested.